src/main/java/io/nats/client/impl/NatsJetStreamManagement.java

```diff
@@ -377,32 +373,30 @@ public LinkedBlockingQueue<MessageInfo> queueMessageBatch(String streamName, Mes
     @Override
     public void requestMessageBatch(String streamName, MessageBatchGetRequest messageBatchGetRequest, MessageInfoHandler handler) throws IOException, JetStreamApiException {
         validateMessageBatchGetRequest(streamName, messageBatchGetRequest);
-        _requestMessageBatch(streamName, messageBatchGetRequest, handler);
+        _requestMessageBatch(streamName, messageBatchGetRequest, true, handler);
     }
 
-    public void _requestMessageBatch(String streamName, MessageBatchGetRequest messageBatchGetRequest, MessageInfoHandler handler) {
+    public void _requestMessageBatch(String streamName, MessageBatchGetRequest messageBatchGetRequest, boolean sendEod, MessageInfoHandler handler) {
         Subscription sub = null;
+
         try {
             String replyTo = conn.createInbox();
             sub = conn.subscribe(replyTo);
 
             String requestSubject = prependPrefix(String.format(JSAPI_DIRECT_GET, streamName));
             conn.publish(requestSubject, replyTo, messageBatchGetRequest.serialize());
 
-            long maxTimeMillis = getTimeout().toMillis();
-            long timeLeft = maxTimeMillis;
-            long start = System.currentTimeMillis();
             while (true) {
-                Message msg = sub.nextMessage(timeLeft);
+                Message msg = sub.nextMessage(getTimeout());
                 if (msg == null) {
                     break;
                 }
                 if (msg.isStatusMessage()) {
                     Status status = msg.getStatus();
                     // Report error, otherwise successful status.
-                    if (status.getCode() < 200 || status.getCode() > 299) {
-                        MessageInfo messageInfo = new MessageInfo(Error.convert(status), true);
-                        handler.onMessageInfo(messageInfo);
+                    if (status.getCode() != Status.EOB) {
+                        handler.onMessageInfo(new MessageInfo(Error.convert(status), true));
+                        sendEod = false; // the error is the EOD since we always end on error
```

Comment:
1. The only time status is not an error is if it is specifically the EOB from the server. We know nothing about any other status functionality, so for now send anything else, even 20x to the user. We need to raise this issue in the go client, because I want to know what other 20x they are expecting

Reply:
2. There is no need to send both an EOD and an error. The error is enough to signify the request has ended.